### PR TITLE
fix: incorrect return type for Model::objectToRawArray()

### PR DESCRIPTION
**Pull request by kenjis**
**Description**
- fix incorrect return types

**Checklist:**
- [x] Securely signed commits
- [] Component(s) with PHPDoc blocks, only if necessary or adds value
- [] Unit testing, with >80% coverage
- [x] User guide updated
- [x] Conforms to style guide

### CODE REVIEW

File: system/BaseModel.php [hunk 1]
```diff
@@ -1678,12 +1678,13 @@ protected function objectToArray($data, bool $onlyChanged = true, bool $recursiv
...
      *
-     * @return array|null Array
+     * @return array Array
      *
      * @throws ReflectionException
      */
-    protected function objectToRawArray($data, bool $onlyChanged = true, bool $recursive = false): ?array
+    protected function objectToRawArray($data, bool $onlyChanged = true, bool $recursive = false): array
     {
+        // @TODO Should define Interface or Class. Enitity has toRawArray() now.
```
Comment (paulbalandan) [inline system/BaseModel.php:1687]:
```suggestion
        // @TODO Should define Interface or Class. Entity has toRawArray() now.
```

File: system/BaseModel.php [hunk 2]
```diff
@@ -1678,12 +1678,13 @@ protected function objectToArray($data, bool $onlyChanged = true, bool $recursiv
      * @param bool          $onlyChanged Only Changed Property
      * @param bool          $recursive   If true, inner entities will be casted as array as well
      *
-     * @return array|null Array
+     * @return array Array
```
Comment (paulbalandan) [inline system/BaseModel.php:1681]:
```suggestion
     * @return array
```

File: system/Model.php [hunk 3]
```diff
@@ -783,11 +783,11 @@ public function update($id = null, $data = null): bool
      * @param object|string $data
      * @param bool          $recursive If true, inner entities will be cast as array as well
      *
-     * @return array|null Array
+     * @return array Array
```
Comment (paulbalandan) [inline system/Model.php:786]:
```suggestion
     * @return array
```

Comment (paulbalandan) [reply system/Model.php:786]: I think this will now be removed by php-cs-fixer

File: system/BaseModel.php (hunk 1 above)
Comment (kenjis) [reply system/BaseModel.php]: Thank you. Fixed.

File: system/Model.php (hunk 3 above)
Comment (kenjis) [reply system/Model.php]: Added the description.

**Review by paulbalandan — APPROVED** [review]

File: user_guide_src/source/changelogs/v4.5.0.rst [hunk 4]
```diff
@@ -24,9 +24,17 @@ Behavior Changes
...
+.. _v450-method-signature-changes:
+
 Method Signature Changes
 ========================
 
+Return Type Changes
+-------------------
+
+- **Model:** The return type of the ``objectToRawArray()`` method in the ``Model``
+  and ``BaseModel`` classes has been changed to ``array``.
```
Comment (MGatner) [inline user_guide_src/source/changelogs/v4.5.0.rst:36]: Save people some research:
```suggestion
  and ``BaseModel`` classes has been changed from ``?array`` to ``array``.
```

**Review by MGatner — CHANGES_REQUESTED** [review]

**Comment by kenjis** [comment]
Rebased to resolve conflicts.

File: system/Model.php [hunk 5]
```diff
@@ -783,11 +783,11 @@ public function update($id = null, $data = null): bool
      * @param object|string $data
      * @param bool          $recursive If true, inner entities will be cast as array as well
      *
-     * @return array|null Array
+     * @return array Array with raw values.
      *
```
Comment (mostafakhudair) [inline system/Model.php:787]:
```suggestion
```
Also here.

File: system/BaseModel.php [hunk 6]
```diff
@@ -1678,12 +1678,13 @@ protected function objectToArray($data, bool $onlyChanged = true, bool $recursiv
      * @param bool          $onlyChanged Only Changed Property
      * @param bool          $recursive   If true, inner entities will be casted as array as well
      *
-     * @return array|null Array
+     * @return array Array with raw values.
      *
```
Comment (mostafakhudair) [inline system/BaseModel.php:1682]:
```suggestion
```
I think the description here is not necessary, since the method has a return type.

Comment (kenjis) [reply system/BaseModel.php:1682]: The intent of the comment is to indicate that the values are **raw values**. 
However, I doubt that most people correctly understand what raw values are.

**Review by MGatner — APPROVED** [review]

**Pull request merged**